app/models/app_remote_kill.rb

```diff
@@ -0,0 +1,65 @@
+class AppRemoteKill < ActiveRecord::Base
+  belongs_to :app
+  belongs_to :split
+
+  attribute :first_bad_version, :app_version
+  attribute :fixed_version, :app_version
+
+  validates :app, :split, :reason, :override_to, :first_bad_version, presence: true
+  validates :reason, uniqueness: { scope: %i(app split) }
+
+  validate :override_to_must_exist
+  validate :fixed_version_must_be_greater_than_first_bad_version
+  validate :must_not_overlap_existing
+
+  scope :affecting, ->(app_build) do
```

Comment:
Where the feature incompletion scope was selecting feature completions satisfied by that app version (i.e. splits not knocked out/assignments not overridden for that version), here the scope selects the remote kills that *do* impact the app. This is because feature completion knockouts operate simply on the boolean absence or presence of satisfying feature completion (and always use `false` as the override variant for knockouts), whereas for remote kills, we need to join in the `override_to` column to know what variant to select (because remote kills can be used on any kind of split).